packages/jazz-tools/src/coValues/coMap.ts

```diff
@@ -564,6 +563,80 @@ export class CoMap extends CoValueBase implements CoValue {
     return cojsonInternals.idforHeader(header, crypto) as ID<M>;
   }
 
+  /**
+   * Given some data, updates an existing CoMap or initialises a new one if none exists.
+   *
+   * @example
+   * ```ts
+   * const activeEvent = await Event.upsertUnique(
+   *   sourceData.identifier,
+   *   workspace.id,
+   *   {
+   *     title: sourceData.title,
+   *     identifier: sourceData.identifier,
+   *     external_id: sourceData._id,
+   *   },
+   *   workspace
+   * );
+   * ```
+   *
+   * @param unique The data that uniquely identifies the CoMap.
+   * @param ownerID The ID of the owner of the CoMap.
+   * @param init The data that the CoMap should contain.
+   * @param creationOptions The creation options for the CoMap.
+   * @param loadOptions The options for loading the CoMap, if it exists (note: `skipRetry` defaults to `true`).
+   * @returns Either an existing & modified CoMap, or a new initialised CoMap if none exists.
+   * @category Subscription & Loading
+   */
+  static async upsertUnique<
+    M extends CoMap,
+    const R extends RefsToResolve<M> = true,
+  >(
+    this: CoValueClass<M>,
+    unique: CoValueUniqueness["uniqueness"],
+    ownerID: ID<Account> | ID<Group>,
+    init: Simplify<CoMapInit<M>>,
+    creationOptions?:
+      | {
+          owner: Account | Group;
+        }
+      | Account
+      | Group,
+    loadOptions?: {
+      resolve?: RefsToResolveStrict<M, R>;
+      loadAs?: Account | AnonymousJazzAgent;
+      skipRetry?: boolean;
```

Comment:
These options look a bit redundant and can be simplified this way:

```ts
    this: CoValueClass<M>,
    options: {
      value: Simplify<CoMapInit<M>>,
      unique: CoValueUniqueness["uniqueness"],
      owner: Account | Group, // required so we don't need ownerId
      resolve?: RefsToResolveStrict<M, R>;
    }
```

Explaination:
- we don't want people to set `skipRetry: false` as per issue description it causes undesired delays
- loadAs can be inferred from `owner._loadedAs`
- we don't need `ownerId` if the owner is passed

Suggesting an options wrapper for everything because with 4 arguments better to make them named.